packages/@aws-cdk/aws-pipes-sources-alpha/lib/kinesis.ts

```diff
@@ -0,0 +1,198 @@
+   * @default - no batching window
+   */
+  readonly maximumBatchingWindow?: Duration;
+
+  /**
+   * (Streams only) Discard records older than the specified age. The default value is -1, which sets the maximum age to infinite. When the value is set to infinite, EventBridge never discards old records.
+   *
+   * Minumum: Duration.seconds(60) (leave undefined to set the maximum age to -1)
+   * Maxiumum: Duration.seconds(604800)
+   *
+   *  @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-maximumrecordageinseconds
+   * @default -1 - maximum age is infinite
+   */
+  readonly maximumRecordAge?: Duration;
+
+  /**
+   * (Streams only) Discard records after the specified number of retries. The default value is -1, which sets the maximum number of retries to infinite. When MaximumRetryAttempts is infinite, EventBridge retries failed records until the record expires in the event source.
+   *
+   * Minumum: -1
+   * Maxiumum: 10000
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-maximumretryattempts
+   * @default -1 (infinite)
+   */
+  readonly maximumRetryAttempts?: number;
+
+  /**
+   * (Streams only) Define how to handle item process failures. {@link OnPartialBatchItemFailure.AUTOMATIC_BISECT} halves each batch and retry each half until all the records are processed or there is one failed message left in the batch.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-onpartialbatchitemfailure
+   * @default off
+   */
+  readonly onPartialBatchItemFailure?: OnPartialBatchItemFailure;
+
+  /**
+   * (Streams only) The number of batches to process concurrently from each shard.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-parallelizationfactor
+   * @default 1
+   */
+  readonly parallelizationFactor?: number;
+
+  /**
+   * (Streams only) The position in a stream from which to start reading.
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-pipes-pipe-pipesourcekinesisstreamparameters.html#cfn-pipes-pipe-pipesourcekinesisstreamparameters-startingposition
+   */
+  readonly startingPosition: KinesisStartingPosition;
+
+  /**
+   * With StartingPosition set to AT_TIMESTAMP, the time from which to start reading, in Unix time seconds.
+   *
+   * @example
+   * '2025-01-01T00:00:00Z'
```

Comment:
@nmussy decided to keep this a string but I added an example here.

Reply:
So just to be clear, `StartingPositionTimestamp` supports both epochs and ISO strings? We should probably amend the documentation above as well, since in only mentions "in Unix time seconds"

Reply:
This is the error if I try epoch:

```[#/SourceParameters/KinesisStreamParameters/StartingPositionTimestamp: [1711726773] is not a valid date-time. Expected [yyyy-MM-dd'T'HH:mm:ssZ, yyyy-MM-dd'T'HH:mm:ss.[0-9]{1,9}Z, yyyy-MM-dd'T'HH:mm:ss[+-]HH:mm, yyyy-MM-dd'T'HH:mm:ss.[0-9]{1,9}[+-]HH:mm]]```

I'm going to open a bug on their documentation.

Reply:
Unless they are considering that "Unix time seconds."

Reply:
I wouldn't think so, it's probably a documentation error yeah. Could you change the JSDoc above then as well?

Reply:
@nmussy I changed this to `in ISO 8601 format`. Submitting AWS doc tickets too.